sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/NullExpressionsSuite.scala

```diff
@@ -87,24 +91,82 @@ class NullExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
-  test("SPARK-16602 Nvl should support numeric-string cases") {
-    def analyze(expr: Expression): Expression = {
-      val relation = LocalRelation()
-      SimpleAnalyzer.execute(Project(Seq(Alias(expr, "c")()), relation)).expressions.head
-    }
-
+  test("coalesce -check return datatype") {
```

Comment:
This test is part of the test case `test("coalesce casts")` in `TypeCoercionSuite`. If you want, you can improve that test case.